src/libpriv/rpmostree-core.c

```diff
@@ -2860,21 +2861,51 @@ handle_file_dispositions (RpmOstreeContext *self,
   return TRUE;
 }
 
+typedef struct {
+  GHashTable *files_skip;
+  GStrv files_remove_regex;
+} FilterData;
+
 static OstreeRepoCheckoutFilterResult
 checkout_filter (OstreeRepo         *self,
                  const char         *path,
                  struct stat        *st_buf,
                  gpointer            user_data)
 {
-  GHashTable *files_skip = user_data;
-  if (g_hash_table_contains (files_skip, path))
-    return OSTREE_REPO_CHECKOUT_FILTER_SKIP;
-  /* Hack for nsswitch.conf: the glibc.i686 copy is identical to the one in glibc.x86_64,
-   * but because we modify it at treecompose time, UNION_IDENTICAL wouldn't save us here. A
-   * better heuristic here might be to skip all /etc files which have a different digest
-   * than the one registered in the rpmdb */
-  if (g_str_equal (path, "/usr/etc/nsswitch.conf"))
-    return OSTREE_REPO_CHECKOUT_FILTER_SKIP;
+  GHashTable *files_skip = ((FilterData*)user_data)->files_skip;
+  GStrv files_remove_regex = ((FilterData*)user_data)->files_remove_regex;
+
+  if (files_skip && g_hash_table_size (files_skip) > 0)
```

Comment:
```suggestion
  if (files_skip)
```

Otherwise, the `nsswitch.conf` hack would now be conditional on that.